### ISSUE TITLE
Newtype types' Lenses are generated without verifying that a `Newtype` instance exists for it and that its constructor is exported

### ISSUE THREAD

**Issue by JordanMartinez** [JordanMartinez/purescript-tidy-codegen-lens#2]
Both of the following modules...
```purescript
module Foo 
  ( MyNewtype -- constructor not exported
  ) where

newtype MyNewtype = MyNewtype Int
deriving instance Newtype MyNewtype _
```

```purescript
module Foo where

newtype MyNewtype = MyNewtype Int
-- no Newtype instance
-- deriving instance Newtype MyNewtype _
```
will produce the following `Lens.purs` file, but the file won't compile:
```purescript
module Foo.Lens where

-- various needed imports here

_MyNewtype :: Lens' MyNewtype Int
_MyNewType = _Newtype
```

**Comment by JordanMartinez** [comment]
There's two checks the code needs to do to implement this correctly.
1. Is the type exported (see #3)
2. Does the type have a derived `Newtype` instance.

The first has yet to be implemented, but there is a design constraint hidden in the second one. Once we ask, "Does some type have a derived `Newtype` instance?", we eventually have to ask, "But does this 'Newtype' type class refer to `Data.Newtype (class Newtype)`? Or something else similarly named?" Consider the following four questions for our decision matrix, illustrated by the next part. 

When we come across a "Newtype" type class instance... 
- Is the type class imported qualified or unqualified?
- Is it imported from the original module or another one that re-exports it?
- Is it an open or closed import?
- If open, is it hiding or showing its members?

For example, below are all the ways one could import the "Newtype" type class using the above matrix:

Unqualified: `derive instance Newtype Type _`
```purescript
import Data.Newtype (class Newtype) -- explicit import - original module
import Data.Newtype hiding (over) -- open import but not hidden - original module
import Data.Newtype -- open import - original module
import Custom.Prelude (class Newtype) -- explicit import - module that re-exports it
import Custom.Prelude hiding (over) -- open import but not hidden - module that re-exports it
import Custom.Prelude -- open import - module that re-exports it
```

Qualified: `derive instance N.Newtype Type _`
```purescript
import Data.Newtype (class Newtype) as N - explicit qualified import - original module
import Data.Newtype hiding (over) as N - open qualified import but not hidden - original module
import Data.Newtype as N -- qualified import - original module
import Custom.Prelude (class Newtype) as N - explicit qualified import - original module
import Custom.Prelude hiding (over) as N - open qualified import but not hidden - original module
import Custom.Prelude as N -- qualified import - original module
```

Due to only knowing the CST for a single file (currently) and not knowing the entire typed module graph in memory, we have no way to verify that the below "Newtype" class refers to the actual `Newtype` class:
```purescript
module Example where
import Custom.Prelude as N
newtype Foo = Foo Int
derive instance N.Newtype Foo _
```
However, if the source file compiles, then we can safely generate a lens for the `Foo` newtype for one reason: **only a few type classes can have their instances derived by the compiler.** If a `derive` clause is used and the type class name is `Newtype`, then it's safe to assume that the "Newtype" class here does refer (ultimately) to `Data.Newtype (class Newtype)`.

**Comment by JordanMartinez** [comment]
Actually, due to the Coercible work, all `Newtype` instances must be derived by the compiler. 

Closed (mostly) in 50e8135015660b9499ce7d01f5e6e39dbc0f433d. Technically, one could still cause `tidy-mklens` to generate a lens for a newtype by importing a type class called `Newtype` (qualified or not) in the file. However, I think this is a rare case in practice and can be fixed when one comes across it.
